eval.py

```diff
@@ -1,81 +1,119 @@
+        if np.sum(gt_mask) == 0:
+            iou_score = np.nan
+        else:
+            iou_score = np.sum(intersection) / (np.sum(union))
 
-    ious = {}
+    return iou_score
+
+
+def get_ious(gt_path, pred_path, true_pos_only):
+    """
+    Returns IoU scores for each combination of CXR and pathology in gt_path and pred_path.
+
+    Args:
+        gt_path (str): path to ground-truth segmentation json file (encoded)
+        pred_path (str): path to predicted segmentation json file (encoded)
+        true_pos_only (bool): if true, TODO
 
-    all_ids = sorted(pred.keys())
+    Returns:
+        ious (dict): dict with 10 keys, one for each pathology (task). Values
+                     are lists of all CXR IoU scores for the pathology key.
+        cxr_ids (list): list of all CXR ids (e.g. 'patient64541_study1_view1_frontal').
+    """
+    with open(gt_path) as f:
+        gt_dict = json.load(f)
+
+    with open(pred_path) as f:
+        pred_dict = json.load(f)
+
+    ious = {}
+    cxr_ids = sorted(gt_dict.keys())
     tasks = sorted(LOCALIZATION_TASKS)
 
     for task in tasks:
         print(f'Evaluating {task}')
         ious[task] = []
 
-        for img_id in tqdm(all_ids):
+        for cxr_id in cxr_ids:
+            # get ground-truth segmentation mask
+            gt_item = gt_dict[cxr_id][task]
+            gt_mask = mask.decode(gt_item)
 
             # get predicted segmentation mask
-            pred_item = pred[img_id][task]
-            pred_mask = mask.decode(pred_item)
-
-            # get ground_truth segmentation mask
-            # if image not in gt, create zero mask
-            if img_id not in gt:
-                gt_mask = np.zeros(pred_item['size'])
+            if cxr_id not in pred_dict:
+                pred_mask = np.zeros(gt_item['size'])
             else:
-                gt_item = gt[img_id][task]
-                gt_mask = mask.decode(gt_item)
+                pred_item = pred_dict[cxr_id][task]
+                pred_mask = mask.decode(pred_item)
 
             assert gt_mask.shape == pred_mask.shape
 
-            # compute metric
-            iou_score = iou_seg(pred_mask, gt_mask)
+            iou_score = calculate_iou(pred_mask, gt_mask, true_pos_only)
             ious[task].append(iou_score)
 
-        assert len(ious[task]) == len(pred.keys())
+        assert len(ious[task]) == len(gt_dict.keys())
 
-    return ious
+    return ious, cxr_ids
 
 
 def bootstrap_metric(df, num_replicates, metric='iou'):
     """
-    Create dataframe of bootstrap samples 
+    Create dataframe of bootstrap samples.
     """
+    # TODO: do we not need to pass in df here?
     def single_replicate_performances():
         sample_ids = np.random.choice(len(df), size=len(df), replace=True)
         replicate_performances = {}
         df_replicate = df.iloc[sample_ids]
 
         for task in df.columns:
+            # TODO: it looks like it's doing the same thing regardless of whether it's miou or pointing game?
```

Comment:
it looks like it's doing the same thing regardless of whether it's miou or pointing game?

Reply:
Yes that is true